### PR TITLE
Develop TIBC cross-chain NFT function

### PR DESCRIPTION
**Pull request by mindcarver**

### CODE REVIEW

File: modules/tibc/apps/nft_transfer/client/cli/flags.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+	FlagAwayFromChain = "awayFromChain"
+)
+
+var (
+	FsNftTransfer    = flag.NewFlagSet("", flag.ContinueOnError)
+)
+
+func init() {
+	FsNftTransfer.String(FlagRelayChain, "", "relay chain used by cross-chain NFT")
+	FsNftTransfer.String(FlagAwayFromChain, "", "indicates whether nft is far away from the source")
```
Comment (chengwenxi) [inline modules/tibc/apps/nft_transfer/client/cli/flags.go:18]: There is no need to declare whether it is IsAwayFromChain or false, we can decide through the NFT class, source_chain, and dest_chain in Keeper.

File: modules/tibc/apps/nft_transfer/client/cli/tx.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
+package cli
+
+import (
+	"fmt"
+	"github.com/bianjieai/tibc-go/modules/tibc/apps/nft_transfer/types"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	"github.com/cosmos/cosmos-sdk/version"
+	"github.com/spf13/cobra"
```
Comment (chengwenxi) [inline modules/tibc/apps/nft_transfer/client/cli/tx.go:10]: make format

File: modules/tibc/apps/nft_transfer/client/cli/tx.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+)
+
+// NewTransferTxCmd returns the command to create a NewMsgTransfer transaction
+func NewTransferTxCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "nft transfer [dest-chain] [class] [id] [uri] [receiver]",
```
Comment (chengwenxi) [inline modules/tibc/apps/nft_transfer/client/cli/tx.go:16]:
```suggestion
		Use:   "nft transfer [dest-chain] [receiver] [class] [id] [uri] ",
```

Comment (chengwenxi) [reply modules/tibc/apps/nft_transfer/client/cli/tx.go:16]: and not need to declare URI, get URI in keeper via NFT class and id

File: modules/tibc/apps/nft_transfer/keeper/relay.go [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+		if err := k.nk.TransferOwner(ctx, data.Class, data.Id, "", data.Uri, "",
+			k.GetNftTransferModuleAddr(types.ModuleName), sender); err != nil{
+			return err
+		}
+
+	} else {
+		// mintNFT
+		if err := k.nk.MintNFT(ctx, data.Class, data.Id, "", data.Uri, "", sender); err != nil{
+			return err
+		}
```
Comment (chengwenxi) [inline modules/tibc/apps/nft_transfer/keeper/relay.go:164]: Don't need to add prefix?

File: modules/tibc/apps/nft_transfer/types/keys.go [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+const (
+	// ModuleName defines the TIBC nft_transfer name
+	ModuleName = "nftTransfer"
+
+	// RouterKey is the message route for the nft-transfer module
+	RouterKey = ModuleName
+
+	// PortID is the default port id that nft-transfer module binds to
+	PortID = "nftTransfer"
```
Comment (chengwenxi) [inline modules/tibc/apps/nft_transfer/types/keys.go:12]: use routingtypes.NFT

File: modules/tibc/apps/nft_transfer/types/msgs.go [hunk 6]
```diff
@@ -0,0 +1,52 @@
+package types
+
+import sdk "github.com/cosmos/cosmos-sdk/types"
+
+const (
+	TypeMsgNftTransfer    = "nft_transfer"
```
Comment (chengwenxi) [inline modules/tibc/apps/nft_transfer/types/msgs.go:6]:
```suggestion
	TypeMsgNftTransfer    = "tibc_nft_transfer"
```

File: modules/tibc/apps/nft_transfer/keeper/relay.go (hunk 4 above)
Comment (mindcarver) [reply modules/tibc/apps/nft_transfer/keeper/relay.go:164]: unnecessary

**Review by chengwenxi — APPROVED** [review]

**Pull request merged**